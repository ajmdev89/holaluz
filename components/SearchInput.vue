<template>
  <div class="flex flex-col w-full items-center">
    <label class="w-full text-white mb-4" for="searchInput">Introduce el CUPS de tu suministro:</label>
    <div class="flex w-full">
      <input id="searchInput" v-model="cups" type="text" autocomplete="off" class="w-full p-2 mr-4">
      <button type="button" aria-label="Buscar" @click="validateCUPSData">
        <img src="/svg/search.svg" alt="" class="w-10 h-10 text-white">
      </button>
    </div>
    <p v-if="isInvalid" class="w-full text-white">Tu suministro no cumple las condiciones para unirse.</p>
  </div>
</template>

<script>
export default {
  name: 'SearchInput',

  data() {
    return {
      cups: '',
      isInvalid: undefined
    }
  },

  methods: {
    async validateCUPSData() {
      if (this.cups) {
        this.isInvalid = await this.$store.dispatch('validateCUPS', this.cups)
        this.cups = ''
      }
    }
  }
}
</script>