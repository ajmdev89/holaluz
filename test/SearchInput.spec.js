import { shallowMount } from '@vue/test-utils'
import SearchInput from '~/components/SearchInput.vue'

describe('SearchInput component -----SearchInput.vue-----', () => {
  const spyValidate = jest.spyOn(SearchInput.methods, 'validateCUPSData')

  it('is a Vue instance', () => {
    const wrapper = shallowMount(SearchInput)
    expect(wrapper.vm).toBeTruthy()
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(SearchInput)
    expect(wrapper).toMatchSnapshot()
  })

  it('it should render', () => {
    const wrapper = shallowMount(SearchInput)

    expect(wrapper.find('label').exists()).toBeTruthy()
    expect(wrapper.find('button').exists()).toBeTruthy()
  })

  it('it should call to validate cups', async () => {
    const wrapper = shallowMount(SearchInput)
    const button = wrapper.find('button')

    await button.trigger('click')

    expect(spyValidate).toHaveBeenCalled()
  })
})
