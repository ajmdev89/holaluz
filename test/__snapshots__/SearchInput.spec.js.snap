// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component -----SearchInput.vue----- matches snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
